### ISSUE TITLE
Properties of the embedded property remain unset

### ISSUE THREAD

**Issue by jaroslavlibal** [doctrine/persistence#317]
Lets say I have the Shop entity with an embedded Location embeddable.

The Location embeddable also has some of its properties embedded - eg. the Address embeddable.

After upgrading from 3.0.4 to 3.1.0, I have all of the attributes of the embedded Address remaining unset.

**Comment by ljmaskey** [comment]
We have noticed the same problem. I spent some time going back and forth between 3.0.4 and 3.1.1 to try and at least track down the difference.

For our situation, at least, it seems to be related to the TypedNoDefaultReflectionProperty. In 3.0.4 it extends PHP's own \ReflectionProperty whose getValue() method does seem to return the expected value. In 3.1.1 that class extends the library's own RuntimeReflectionProperty (which then extends PHP's own \ReflectionProperty). At line 43 in the RuntimeReflectionProperty, it seems as though the value for $this->key has a leading backslash on the class name, but only in the case for embedded entities that are within another embedded entity.

I've not dug any deeper to work out if that is on purpose, or why exactly it is happening. However as a really hopeful (and naive) attempt at seeing some change I added:
        `$class = ltrim($class, '\\');`
as the first line of the constructor for RuntimeReflectionProperty and while our local tests are far from exhaustive, they have started passing at least.

I'm not sure if that is a valid solution or if it's simply wallpapering over the real issue. Either way I hope that it is information that's more useful than not...
